runtime/Primitive_lazy.res

```diff
@@ -34,10 +34,10 @@ type t<'a> = {
 
 %%private(external fnToVal: (unit => 'a) => 'a = "%identity")
 %%private(external valToFn: 'a => unit => 'a = "%identity")
-%%private(external castToConcrete: lazy_t<'a> => t<'a> = "%identity")
```

Comment:
I don't think we can make `lazy_t` and `t` the same here, because `lazy_t` needs to be covariant and `t` is mutable. Also, the naming might change for clarity: currently `lazy_t` represents the type `Lazy.t` while `t` represents an internal implementation-only type used to generate specific js code (and cannot be exported as it would be unsound to export the mutable part anyway).

Reply:
Overall, bringing `lazy_t` out of buildings seems to bring clarity, as it makes a few things more explicit.